catalog/app/containers/Bucket/Queries/Athena/Athena.tsx

```diff
@@ -237,23 +237,17 @@ function ResultsContainer({ className }: ResultsContainerProps) {
           <SeeDocsForCreatingPackage />
         )}
       </ResultsBreadcrumbs>
-      {/* eslint-disable-next-line no-nested-ternary */}
-      {results.data.rows.length ? (
+      {!results.data.rows.length && execution?.error ? (
+        <Alert
+          error={execution?.error || new Error("Couldn't fetch query results")}
```

Comment:
`execution?.error` here can't be nullish because of the condition above, so you don't need the fallback.
i can't quite wrap my head around what this condition is supposed to mean -- only show the error if there's no data?

Reply:
Thank you, I examined the code deeper, and found out this condition will never happen. `execution.error` could be only inside executions list

Reply:
Fixed